BaseUI/src/components/EditButton.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import TextButton from './TextButton';
+
+// This component's functioning like a wrapper
+class EditButton extends React.Component<
+  {
+    text: string;
+    color: string;
+    align: 'flex-start' | 'center';
+    textAlign?: 'center' | 'auto';
+    editMode: boolean;
+    onPress?: ((text: boolean) => void) | (() => void);
+  },
+  {}
+> {
+  render(): React.ReactNode {
+    if (!this.props.editMode) {
```

Comment:
a component should be responsible for its task. So, a button should show a button. To show it or not should decide on a header or some other components that are above in the hierarchy.